SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialActorUtils.h

```diff
@@ -44,6 +50,23 @@ inline FString GetConnectionOwningWorkerId(const AActor* Actor)
 	return FString();
 }
 
+template <typename Functor>
+void ForeachReplicatedActorInHierarchy(AActor* HierarchyActor, Functor&& InFunctor)
+{
+	if (HierarchyActor)
+	{
+		if (HierarchyActor->GetIsReplicated())
+		{
+			InFunctor(HierarchyActor);
+		}
+
+		for (AActor* Child : HierarchyActor->Children)
+		{
+			ForeachReplicatedActorInHierarchy(Child, InFunctor);
```

Comment:
Nit - should forward the functor.

Reply:
Wouldn't that run the risk of moving it on one branch, and having a default/invalid one for other children ?

Reply:
Oh yeah, it's in a loop. In that case it should take a const lvalue ref?